utils/hashing/hashing.go

```diff
@@ -9,7 +9,11 @@ import (
 	"fmt"
 	"io"
 
-	"golang.org/x/crypto/ripemd160"
+	// This algorithm does not represent a security risk since it is
```

Comment:
Why does this prove this isn't a security risk?

Reply:
I guess what we should say is something like "this is what is used in bitcoin and other cryptocurrencies", but the fact that we use this to convert a public key to an address doesn't mean this is inherently secure. 

For instance, suppose that ripemd160 would map half of its possible inputs to the same output. Then, it wouldn't be secure, because you could steal funds from every second address (on average).

Reply:
I think we should say, that we only use ripemd160 to derive addresses, in the same way Bitcoin does, and thus in this aspect we're as secure as Bitcoin is. 

I just commented because address derivation has security importance, so we should be precise in reasoning we put as comments :-) 

@StephenButtolph thoughts?

Reply:
Yeah I think it would make sense to essentially say:
1. ripemd160 is also used by bitcoin in a similar way to generate an address from a public key.
2. ripemd160 is not generally recommended for use. However, it is only used to hash small payloads which is generally harder to attack than large payload size collisions.

Reply:
We could also link to: [this paper](https://online.tugraz.at/tug_online/voe_main2.getvolltext?pCurrPk=17675) which is I beleive the most up-to-date research on collision resistance of ripemd160.